rust/src/http2/http2.rs

```diff
@@ -809,6 +809,7 @@ impl HTTP2State {
                                     let mut tx_same = &mut self.transactions[index - 1];
                                     tx_same.ft.tx_id = tx_same.tx_id - 1;
                                     let (files, flags) = self.files.get(dir);
+                                    tx_same.tx_data.init_files_opened(); // TODO do we have a place for doing this once?
```

Comment:
@catenacyber I did not field a single place where we can initialize a file in the tx. This works because it just keeps the file cnt at 1, but its called for each chunk of file data.

Reply:
Indeed, there is no initialization, just each chunk of HTTP2 `DATA` is part of the file 
Would you like some condition like `if isFirstDataChunk(files)` to run this line ?

Reply:
Yeah I think this would be useful

Reply:
Guess it would be `if files.tail == ptr::null_mut() {`
Or `self.ft.file_open` but then, maybe `ft` needs to ne duplicated in `ft_tc` and `ft_ts`

Reply:
I think we need that `ft_ts` and `ft_tc` anyway, so `self.ft.file_open` might work.

Reply:
So, how do you want to proceed, should I create a PR on top of these commits ?